R/tmiic.utils.R

```diff
@@ -0,0 +1,1343 @@
+  n_layers, delta_t, movavg, keep_max_data, max_nodes)
+  {
+  # Check number of layers parameter
+  #
+  if ( ! is.null (n_layers) )
+    {
+    if ( test_param_wrong_int (n_layers, min=2, max=NA) )
+      {
+      if ( "n_layers" %in% colnames(state_order) )
+        miic_warning ("parameters", "supplied value ", list_to_str (n_layers),
+          " for the number of layers is invalid,",
+          " if not NULL, it must be an integer >= 2.",
+          " This issue has no impact as the number of layers is provided",
+          " in the state_order.")
+      else
+        miic_warning ("parameters", "supplied value ", list_to_str (n_layers),
+          " for the number of layers is invalid,",
+          " if not NULL, it must be an integer >= 2.",
+          " The number of layers will be estimated from the data.")
+      }
+    else # valid n_layers
+      {
+      if ( ! ("n_layers" %in% colnames(state_order)) )
+        {
+        state_order$n_layers = n_layers
+        state_order$n_layers[state_order$is_contextual == 1] = 1
+        }
+      else # n_layers in state_order
+        {
+        na_in_so = is.na (state_order$n_layers)
+        if ( any (na_in_so) )
+          {
+          miic_warning ("parameters", "the number of layers is supplied both",
+            " in the state_order and as parameter. As some values are missing",
+            " in the state_order, the parameter will be used to fill these",
+            " missing values.")
+          state_order$n_layers[na_in_so & (state_order$is_contextual == 0)] = n_layers
+          }
+        else
+          miic_warning ("parameters", "the number of layers is supplied both",
+            " in the state_order and as parameter. The parameter will be",
+            " ignored.")
+        }
+      }
+    }
+  #
+  # Check delta_t
+  #
+  if ( ! is.null (delta_t) )
+    {
+    if ( test_param_wrong_int (delta_t, min=1, max=NA) )
+      {
+      if ( "delta_t" %in% colnames(state_order) )
+        miic_warning ("parameters", "supplied value ", list_to_str (delta_t),
+          " for the delta t parameter is invalid,",
+          " if not NULL, it must be an integer >= 1.",
+          " This issue has no impact as the delta t is provided",
+          " in the state_order.")
+      else
+        miic_warning ("parameters", "supplied value ", list_to_str (delta_t),
+          " for the delta t parameter is invalid,",
+          " if not NULL, it must be an integer >= 1.",
+          " The delta t will be estimated from the data.")
+      }
+    else # valid delta_t
+      {
+      if ( ! ("delta_t" %in% colnames(state_order)) )
+        {
+        state_order$delta_t = delta_t
+        state_order$delta_t[state_order$is_contextual == 1] = 0
+        }
+      else # delta_t in state_order
+        {
+        na_in_so = is.na (state_order$delta_t)
+        if ( any (na_in_so) )
+          {
+          miic_warning ("parameters", "the delta t is supplied both",
```

Comment:
EN (OD)

Reply:
is both supplied in the state_order and as a parameter.